techniques/system/inventory/1.0/fusionAgent.st

```diff
@@ -427,49 +427,79 @@ bundle agent turnUsersToUnicode
       "CFKEY_cmd"      string => "${sys.winsysdir}\cmd.exe /c \"type \"${sys.workdir}\ppkeys\localhost.pub\"\"";
+    windows.cygwin::
+      "CFKEY_cmd"      string => "/usr/bin/cat ${sys.workdir}/ppkeys/localhost.pub";
+    android::
+      "CFKEY_cmd"      string => "/system/bin/cat ${sys.workdir}/ppkeys/localhost.pub";
+    !windows.!android::
+      "CFKEY_cmd"      string => "/bin/cat ${sys.workdir}/ppkeys/localhost.pub";
+    any::
+      "CFKEY"          string => execresult("${CFKEY_cmd}", "noshell");
+
+    # define USER
+    windows.!cygwin::
       "USER"           string => getenv("USERNAME", 40);
-      "RUDDERUUID_cmd" string => "${sys.winsysdir}\cscript.exe /Nologo \"${g.rudder_tools}/uuid.vbs\"";
-      "polserv_uuid"   string => readfile( "${sys.workdir}\rudder-server-uuid.txt" , "50" );
-      # Somehow, using the variable rather than the path fails with readstringlist
-      "users" slist => { readstringlist("C:\Program Files\Rudder\var\tmp\UserList.list","#.*","[\n| |\r]",50,4000) };
-&endif&
+    windows.cygwin::
+      "USER"           string => execresult("/usr/bin/whoami", "noshell");
+    android::
+      "USER"           string => execresult("/system/xbin/whoami", "noshell");
+    solaris::
+      "USER"           string => execresult("/usr/bin/id | /usr/bin/cut -d\( -f2 | /usr/bin/cut -d\) -f1", "noshell");
+    !windows.!android.!solaris::
+      "USER"           string => execresult("/usr/bin/whoami", "noshell");
 
-    cygwin::
-      "mywinpath"      string => execresult("/usr/bin/echo $WINDIR ", "useshell");
-      "CFKEY_cmd"      string => "/usr/bin/cat ${sys.workdir}/ppkeys/localhost.pub";
-      "USER_cmd"       string => "/usr/bin/whoami";
-      "RUDDERUUID_cmd" string => "${sys.winsysdir}\cscript.exe /Nologo ${g.rudder_tools}/uuid.vbs";
-      "polserv_uuid"   string => readfile( "${sys.workdir}/rudder-server-uuid.txt" , "50" );
-      "users"          slist => { readstringlist("${inventory.UserListFile}","#.*","[\n| |\r]",10,4000) };
+    # define RUDDERUUID
+    windows.!cygwin::  
+      "RUDDERUUID"     string => execresult("${sys.winsysdir}\cscript.exe /Nologo \"${g.rudder_tools}/uuid.vbs\"", "noshell");
+    windows.cygwin::
+      "RUDDERUUID"     string => execresult("${sys.winsysdir}\cscript.exe /Nologo ${g.rudder_tools}/uuid.vbs", "noshell");
+    android::
+      "RUDDERUUID"     string => execresult("/system/xbin/sqlite3 /data/data/com.android.providers.settings/databases/settings.db \"select value from secure where name = 'android_id'\"", "noshell");
+    aix::
+      "RUDDERUUID"     string => execresult("/usr/bin/uname -f", "noshell");
+    solaris::
+      "RUDDERUUID"     string => execresult("smbios | grep UUID | sed 's/  UUID: //'", "useshell");
+    freebsd::
+      "RUDDERUUID"     string => execresult("/usr/local/sbin/dmidecode -s system-uuid", "noshell");
+    linux.!xen::
+      "RUDDERUUID"     string => execresult("/usr/sbin/dmidecode -s system-uuid", "noshell");
+    linux.xen::
```

Comment:
This is missing the class xenrudderuid (and a note explaining that this comes from a regex evaluated in the classes: section below)